compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -317,6 +327,24 @@ execCompile inputFile outputFile opts = withProjectRoot' (ProjectOpts Nothing (P
         createDirectoryIfMissing True $ takeDirectory outputFile
         B.writeFile outputFile $ Archive.encodeArchive bs
 
+execLint :: FilePath -> Options -> Command
+execLint inputFile opts =
+  withProjectRoot' (ProjectOpts Nothing (ProjectCheck "" False)) $ \relativize ->
+  do
+    loggerH <- getLogger opts "lint"
+    inputFile <- toNormalizedFilePath <$> relativize inputFile
+    opts' <- mkOptions opts
+    defaultHlintDataDir <-locateRunfiles $ mainWorkspace </> "compiler/damlc/daml-ide-core"
+    -- If not `--with-hlint path` given on the command line, assume
+    -- the testing default.
+    let opts'' =
```

Comment:
This might be easier to read if you shadow `opts` (you’ll have to use `opts <- pure …` here) instead of having both `opts`, `opts'', and `opts''` in scope at the same time.

Reply:
Right you are. I'll give it another pass.

Reply:
✔️ ... pushing this now. May I prevail upon you to have another look?